### PR TITLE
refactor: adapt avatar widget

### PR DESCRIPTION
**Pull request by imGok**
https://user-images.githubusercontent.com/43041967/232571054-5782d5d3-0df4-47d7-89a6-9b7de68b9a24.mp4

### CODE REVIEW

File: lib/widgets/avatar.dart [hunk 1]
```diff
@@ -42,55 +49,47 @@ class Avatar extends StatelessWidget {
...
+        borderRadius: BorderRadius.circular((size ?? 40) * 0.5),
+        border: Border.all(color: Colors.transparent),
+        boxShadow: boxShadows ?? [],
+        gradient: LinearGradient(
+          begin: Alignment.topCenter,
+          end: Alignment.bottomCenter,
+          tileMode: TileMode.decal,
+          colors: [
+            name?.lightColorAvatarGradientStart ?? defaultGradientColorStart,
+            name?.lightColorAvatarGradientEnd ?? defaultGradientColorEnd,
```
[the remark this answers is not in the document]
Comment (imGok) [reply lib/widgets/avatar.dart:77]: There is a gradient in the mockup 
image
I just followed the design and the CSS related 
image

**Comment by hoangdat** [comment]
- improve gradient color for avatar

**Comment by hoangdat** [comment]
please do https://github.com/linagora/linagora-design-flutter/issues/2 first

File: .vscode/settings.json
Comment (hoangdat) [inline .vscode/settings.json:1]: we should ignore all IDE files

File: ios/FluffyChat Share/FluffyChat Share.entitlements [hunk 2]
```diff
@@ -3,8 +3,6 @@
 <plist version="1.0">
 <dict>
 	<key>com.apple.security.application-groups</key>
-	<array>
-		<string>group.im.fluffychat.app</string>
-	</array>
```
Comment (hoangdat) [inline ios/FluffyChat Share/FluffyChat Share.entitlements:8]: remove

File: ios/Runner.xcodeproj/project.pbxproj
Comment (hoangdat) [inline ios/Runner.xcodeproj/project.pbxproj:1]: remove

File: ios/Runner/Runner.entitlements
Comment (hoangdat) [inline ios/Runner/Runner.entitlements:1]: remove

File: lib/utils/string_extension.dart [hunk 3]
```diff
@@ -12,4 +17,42 @@ extension StringCasingExtension on String {
...
+    if (isNotEmpty) {
+      final codeUnits = this.codeUnits;
+      if (codeUnits.isNotEmpty) {
+        final sumCodeUnits = codeUnits.sum;
+        final index = sumCodeUnits % AppColor.mapGradientColor.length;
+        return index;
+      }
+    }
+    return 0;
+  }
```
Comment (hoangdat) [inline lib/utils/string_extension.dart:57]: Duplicated with Tmail. How to do it better?

File: linux/flutter/generated_plugin_registrant.cc
Comment (hoangdat) [inline linux/flutter/generated_plugin_registrant.cc:1]: remove

File: linux/flutter/generated_plugins.cmake
Comment (hoangdat) [inline linux/flutter/generated_plugins.cmake:1]: remove

File: scripts/build-ios.sh [hunk 4]
```diff
@@ -1,5 +1,5 @@
 #!/usr/bin/env bash
-git apply ./scripts/enable-android-google-services.patch
+#git apply ./scripts/enable-android-google-services.patch
```
Comment (hoangdat) [inline scripts/build-ios.sh:2]: don't touch before we really understand what happen

**Comment by hoangdat** [comment]
image

**Comment by hoangdat** [comment]
image

File: lib/utils/string_extension.dart [hunk 5]
```diff
@@ -12,4 +12,26 @@ extension StringCasingExtension on String {
...
+    if (first.isNotEmpty) {
+      codeUnits.writeCharCode(first.runes.first);
+    }
+    
+    if (last.isNotEmpty) {
+      codeUnits.writeCharCode(last.runes.first);
+    }
+
+    return codeUnits.toString().toUpperCase();
+  }
```
Comment (hoangdat) [inline lib/utils/string_extension.dart:36]: IMO, you should add unit-tests for this method

**Comment by sherlockvn** [comment]
> image

Simulator Screenshot - iPhone 14 - 2023-04-17 at 02 18 05
Fixed it!

**Comment by sherlockvn** [comment]
> image

What is the problem ?

File: pubspec.lock [hunk 6]
```diff
@@ -2301,5 +2310,5 @@ packages:
     source: hosted
     version: "3.1.1"
 sdks:
-  dart: ">=2.19.0 <3.0.0"
+  dart: ">=2.19.3 <3.0.0"
```
Comment (hoangdat) [inline pubspec.lock:2313]: Mạnh said that, our CI/CD is not compatible with 2.19.3, please carefull

File: pubspec.yaml [hunk 7]
```diff
@@ -94,6 +93,10 @@ dependencies:
   vrouter: ^1.2.0+21
   wakelock: ^0.6.1+1
   webrtc_interface: ^1.0.10
+  linagora_design_flutter:
+    git: 
+      url: git@github.com:linagora/linagora-design-flutter.git
+      ref: feature/add-linagora-circle-avatar
```
Comment (hoangdat) [inline pubspec.yaml:99]: please update

**Review by hoangdat — APPROVED** [review]

**Pull request merged**